tests/unit/acquisition/test_function.py

```diff
@@ -85,14 +90,39 @@ def test_expected_improvement_builder_builds_expected_improvement(
     npt.assert_array_almost_equal(acq_fn(query_at), expected)
 
 
-def test_expected_improvement() -> None:
-    def _ei(x: tf.Tensor) -> tf.Tensor:
-        n = tfp.distributions.Normal(0, 1)
-        return - x * n.cdf(-x) + n.prob(-x)
+@random_seed()
+@pytest.mark.parametrize('best, rtol, atol', [
+    (tf.constant([50.0]), 0.01, 1e-3),
+    (BRANIN_GLOBAL_MINIMUM[None], 0.02, 1e-3),
+    (BRANIN_GLOBAL_MINIMUM[None] * 1.01, 0.02, 1e-3)
+])
+@pytest.mark.parametrize('variance_scale, num_samples_per_point', [
+    (0.1, 1000), (1.0, 50_000), (10.0, 1_000_000), (100.0, 1_000_000)
```

Comment:
More of an observation than a specific suggestion:

Some of these parameters only really serve to make sure that the approximation method used by the test is sufficiently accurate to be within the tolerance of the true value. So they're not really specifying the test case, they're provided some extra implementation detail needed to make the test work. 

The test may be be more readable if this wasn't the case, but I can't think of a way to do this which wouldn't increase the runtime a lot, or introduce another (probably worse) form of complexity.

Reply:
yes i know what you mean. i tend to use `parametrize` for all sorts of things, even parametrizing over which function to test, so I think this is OK, though perhaps not common